addons/api/addon/adapters/application.js

```diff
@@ -125,12 +125,8 @@ export default class ApplicationAdapter extends RESTAdapter.extend(
       //add the result items to a data array
       if (result && result.items) {
         data.push(...result.items);
-        //pass in the refresh token for subsequent calls to fetch the remaining list items
-        query.refresh_token = result.refresh_token;
-        //break the loop as soon the response_type becomes complete
-        if (result.response_type === 'complete') {
```

Comment:
@ZedLi thanks for bringing this up. We don't need this. I initially added it as I ran into infinite loops, but the `while` condition handles that.